### PR TITLE
fix(a380x/mfd): Fixed DIRECT TO selection of active waypoint does nothing

### PR DESCRIPTION
**Pull request by sognodelx**
## Summary of Changes

DIRECT TO selection of active waypoint results in index 0 and thus onModified leads to no action. Changed the condition to check for null in the onModified action to allow selection of active waypoint.

## Screenshots (if necessary)

image

## References

image

## Additional context

Discord username (if different from GitHub):

## Testing instructions

In flight with an active flight plan select the active waypoint as DIRECT TO.

## How to download the PR for QA

Every new commit to this PR will cause new A32NX and A380X artifacts to be created, built, and uploaded.

1. Make sure you are signed in to GitHub
1. Click on the **Checks** tab on the PR
1. On the left side, find and click on the **PR Build** tab
1. Click on either **flybywire-aircraft-a320-neo**, **flybywire-aircraft-a380-842 (4K)** or **flybywire-aircraft-a380-842 (8K)** download link at the bottom of the page

### CODE REVIEW

**Review by Benjozork — APPROVED** [review]

**Comment by sognodelx** [comment]
Sorry, had to force push the commit again so that the correct e-mail address is set (no changes in content).

File: fbw-a380x/src/systems/instruments/src/MFD/pages/FMS/F-PLN/MfdFmsFplnDirectTo.tsx [hunk 1]
```diff
@@ -164,7 +164,7 @@ export class MfdFmsFplnDirectTo extends FmsPage<MfdFmsFplnDirectToProps> {
                     containerStyle="width: 175px;"
                     alignLabels="flex-start"
                     onModified={(i, text) => {
-                      if (i) {
+                      if (i != null) {
```
Comment (flogross89) [inline fbw-a380x/src/systems/instruments/src/MFD/pages/FMS/F-PLN/MfdFmsFplnDirectTo.tsx:167]:
```suggestion
                      if (i !== null && i !== undefined) {
```
Thanks for the fix, well spotted! We should avoid juggling checks, as they could be perceived as unintentional. I'd rather check for null and undefined explicitly

**Review by flogross89 — CHANGES_REQUESTED** [review]

**Comment by nebj00la** [comment]
Quality Assurance Trainee Report

Discord Username  : nebj00la
Object of testing : #9438
Aircraft          : A380X
Tier of Testing   : 1
Date              : 07/11/2024

Testing Process: 

1. Loaded various waypoints for approach to KBOS.
2. Flew HDG after a new waypoint was set, then loaded DIRECT to the same waypoint.

Testing Results: 
Passed

Media:

**Review by flogross89 — APPROVED** [review]

File: fbw-a380x/src/systems/instruments/src/MFD/pages/FMS/F-PLN/MfdFmsFplnDirectTo.tsx (hunk 1 above)
Comment (sognodelx) [reply fbw-a380x/src/systems/instruments/src/MFD/pages/FMS/F-PLN/MfdFmsFplnDirectTo.tsx]: Saw it in the live stream with capta320 and riiiva, in which you were also present. I asked myself exactly whether I should do an explicit null check. Thx for the feedback, even if it is a small change.

Comment (flogross89) [reply fbw-a380x/src/systems/instruments/src/MFD/pages/FMS/F-PLN/MfdFmsFplnDirectTo.tsx]: Sure, thanks for the fix. We will include it in version 0.12.2, which we're going to release soon.

**Pull request merged**